File: TransCoder_tokenized_test_set_functions/python/ENTRINGER_NUMBER_1.py
def zigzag ( n, k ) :
    dp = [ [ 0 for x in range ( k + 1 ) ] for y in range ( n + 1 ) ]
    dp [ 0 ] [ 0 ] = 1
    for i in range ( 1, n + 1 ) :
        dp [ i ] [ 0 ] = 0
    for i in range ( 1, n + 1 ) :
        for j in range ( 1, k + 1 ) :
            dp [ i ] [ j ] = ( dp [ i ] [ j - 1 ] + dp [ i - 1 ] [ i - j ] )
    return dp [ n ] [ k ]

File: TransCoder_tokenized_test_set_functions/java/ENTRINGER_NUMBER_1.java
static int zigzag ( int n, int k ) {
  int dp [ ] [ ] = new int [ n + 1 ] [ k + 1 ];
  dp [ 0 ] [ 0 ] = 1;
  for ( int i = 1;
  i <= n;
  i ++ ) dp [ i ] [ 0 ] = 0;
  for ( int i = 1;
  i <= n;
  i ++ ) {
    for ( int j = 1;
    j <= Math . min ( i, k );
    j ++ ) dp [ i ] [ j ] = dp [ i ] [ j - 1 ] + dp [ i - 1 ] [ i - j ];
  }
  return dp [ n ] [ k ];
}